src/components/SignInPage/SigninPage.js

```diff
@@ -0,0 +1,229 @@
+        });
+      } else {
+        setAlert({
+          text: e.message,
+          opened: true
+        });
+      }
+    });
+  }
+
+  const authorise = async () => {
+    if (!login) {
+      setAlert({
+        text: 'Please, print your login.',
+        opened: true
+      });
+      return;
+    }
+
+    if (!password) {
+      setAlert({
+        text: 'Please, print your password.',
+        opened: true
+      });
+      return;
+    }
+
+    await axios.post('http://localhost:8000/authorise', {
+      login,
+      password,
+    }, {
+      withCredentials: true, 
+      credentials: 'include'
+    }).then(result => {
+      const { data } = result;
+      setIsAuth(true);
+      localStorage.setItem('token', data.accessToken);
+      history.push('/mainPage');
+    }).catch(e => {
+      setAlert({
+        text: e.message,
+        opened: true
+      });
+    });
+  }
+
+  return (
+    <div className="signin-page-wrapper">
+      <div className="signin-central-window">
+        <header>
+          {isRegistration
+            ? 'Register'
+            : 'Sign In'
+          }
+        </header>
+        <section>
+          <div className="signin-input-wrapper">
+            <Label
+              for="login"
+              size="lg"
+            >
+              Login
+            </Label>
+            <Input
+              id="login"
+              name="login"
+              size="lg"
+              placeholder="Enter your login, please"
+              value={login}
+              onChange={(e) => setLogin(e.currentTarget.value)}
+              type="text"
+            />
+          </div>
+          <div className="signin-input-wrapper">
+            <Label
+              for="password"
+              size="lg"
+            >
+              Password
+            </Label>
+            <Input
+              id="password"
+              name="password"
+              size="lg"
+              placeholder="Enter your password, please"
+              value={password}
+              onChange={(e) => setPassword(e.currentTarget.value)}
+              type="password"
+            />
+          </div>
+          {
+            isRegistration &&
+            <div className="signin-input-wrapper">
+              <Label
+                for="repeated-password"
+                size="lg"
+              >
+                Repeate your password
+              </Label>
+              <Input
+                id="repeated-password"
+                name="repeated-password"
+                size="lg"
+                placeholder="Repeate your password, please"
+                value={repeatedPassword}
+                onChange={(e) => setRepeatedPassword(e.currentTarget.value)}
+                type="password"
+              />
+            </div>
+          }
+        </section>
+        <footer>
+          <Button
+            color="primary"
+            size="lg"
+            onClick={
+              () => isRegistration
```

Comment:
try to avoid conditions like this in JSX it hard to read, better to move it to separate func, and call it instead.